<!-- This file creates a simple path and runs the car against the path in Gazebo. -->

<launch>
  <node pkg="tf2_ros" type="static_transform_publisher" name="fake_mapper"
        args="0 0 0 0 0 0 1 map odom" />
  <node pkg="map_server" type="map_server" name="map_server" args="$(find lab5)/data/sim_tunnels.yaml" required="true" />

  <!-- use rviz as the gui instead of gazebo -->
  <include file="$(find racecar_gazebo)/launch/racecar.launch">
    <arg name="world_name" value="racecar_tunnel" />
    <arg name="gui" value="false" />
  </include>
  <node pkg="rviz" type="rviz" name="rviz" args="-d '$(find lab5)/rviz/path_follow.rviz'" required="true" />

  <!--node pkg="lab6" type="path_tracker.py" name="path_tracker" /-->

  <node pkg="lab5" name="path_follower_node" type="path_follower.py">
    <remap from='~path' to='/path_follower/robot_path' />
  </node>

  <!-- Create a default test path -->
  <node pkg="lab5" name="test_path_source" type="test_path_source.py">
    <remap from='~robot_path' to='/robot_path' />
  </node>

  <node pkg="lab6" type="not_rrt.py" name="rrt_planner">
    <remap from="~path" to="/path_follower/robot_path" />
  </node>

  <node pkg="lab6" type="top_planner.py" name="top_planner" output="screen">
    <remap from="~target_pose" to="/rrt_planner/goal" />
    <remap from="~map" to="map" />
    <remap from="~scan" to="scan" />
    <remap from="~path" to="/robot_path" />
    <remap from="~marker_poses" to="/marker_poses" />
  </node>

</launch>
